### PR TITLE
libsigc++

### PR DESCRIPTION
**Pull request by scopatz**
Not sure how to specify that a package does not rely on Python or that it needs a C++ compiler. Testing that here to some degree.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@scopatz if you add `[skip appveyor]` to your commit messages you will leave AppVeyor free for the rest of conda-forge :wink:

File: recipes/libsigcpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  url: http://ftp.acc.umu.se/pub/GNOME/sources/libsigc++/{{ major_minor }}/libsigc++-{{ version }}.tar.xz
+  sha256: fdace7134c31de792c17570f9049ca0657909b28c4c70ec4882f91a03de54437
+
+build:
+  number: 0
+    skip: true  # [win]
+
+requirements:
+  build:
+  run:
```
Comment (ocefpaf) [inline recipes/libsigcpp/meta.yaml]: You can remove these if they are empty.

File: recipes/libsigcpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  number: 0
+    skip: true  # [win]
+
+#requirements:
+#  build:
+#  run:
+
+#test:
+#  imports:
+#  commands:
```
Comment (ocefpaf) [inline recipes/libsigcpp/meta.yaml]: The linter will complain if there is no tests. You can do a "exist lib test" like [this one](https://github.com/conda-forge/hdfeos2-feedstock/blob/master/recipe/meta.yaml#L26).

**Comment by scopatz** [comment]
> @scopatz if you add [skip appveyor] to your commit messages you will leave AppVeyor free for the rest of conda-forge :wink: 

Good to know.  Is there documentation on this somewhere?

File: recipes/libsigcpp/build.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+echo  $PKG_CONFIG
+PKG_CONFIG_PATH=
+echo  $PKG_CONFIG_LIBDIR
+echo $ACLOCAL_FLAGS
+echo 'int main(){return 0;}'>examples/hello_world.cc
+
+# configure, make, install
+./configure  --prefix=$PREFIX CFLAGS='-g -O2' CXXFLAGS='-g -O2' CPP= CXX=
+make
+make install
```
Comment (ocefpaf) [inline recipes/libsigcpp/build.sh]: Is there a `make check`?

**Comment by ocefpaf** [comment]
> Good to know. Is there documentation on this somewhere?

Nope. We are like caveman sharing knowledge via demonstrations. (But we are putting together a conda-forge cheat-sheet and that should be in there.)

**Comment by scopatz** [comment]
> Nope. We are like caveman sharing knowledge via demonstrations. (But we are putting together a conda-forge cheat-sheet and that should be in there.)

haha nice. I look forward to developing a written tradition.

File: recipes/libsigcpp/build.sh (hunk 3 above)
Comment (scopatz) [inline recipes/libsigcpp/build.sh]: Yes, just added.

File: recipes/libsigcpp/meta.yaml (hunk 2 above)
Comment (scopatz) [inline recipes/libsigcpp/meta.yaml]: Added.

File: recipes/libsigcpp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  fn: libsigc++-{{ version }}.tar.xz
+  url: http://ftp.acc.umu.se/pub/GNOME/sources/libsigc++/{{ major_minor }}/libsigc++-{{ version }}.tar.xz
+  sha256: fdace7134c31de792c17570f9049ca0657909b28c4c70ec4882f91a03de54437
+
+build:
+  number: 0
+  skip: true  # [win]
+
+test:
+  command:
```
Comment (jakirkham) [inline recipes/libsigcpp/meta.yaml]: Needs an `s`.

**Comment by scopatz** [comment]
So how do we get a compiler with C++11? This seems to be what it is failing on now.

**Comment by scopatz** [comment]
Many of the projects I want to add now require this.

File: recipes/libsigcpp/build.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+echo  $CXXCPP
+echo  $M4
+echo  $PERL
+echo  $DOT
+echo  $DOXYGEN
+echo  $XSLTPROC
+echo  $PKG_CONFIG
+PKG_CONFIG_PATH=
+echo  $PKG_CONFIG_LIBDIR
+echo $ACLOCAL_FLAGS
```
Comment (jakirkham) [inline recipes/libsigcpp/build.sh]: No environment variables will be set. So this can all go.

File: recipes/libsigcpp/build.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+echo 'int main(){return 0;}'>examples/hello_world.cc
+
+# configure, make, install, check
+./configure  --prefix=$PREFIX CFLAGS='-g -O2' CXXFLAGS='-g -O2' CPP= CXX=
```
Comment (jakirkham) [inline recipes/libsigcpp/build.sh]: Though we may change that in the future for 32 and 64 bit support. So, you might want to allow the old `CFLAGS` and `CXXFLAGS` in for that rare case we do set them.

**Comment by jakirkham** [comment]
> So how do we get a compiler with C++11? This seems to be what it is failing on now.

On Linux, we have to install the `gcc` package as a build time dependency and `libgcc` as a run time dependency. We are debating different ways to include such a compiler on the image. There is no decision yet. Though I can point you to the discussion.

On Mac, we use `clang`.

Also, you may need some extra flags. Take a look at this [`build.sh`]( https://github.com/conda-forge/boost-feedstock/blob/master/recipe/build.sh ).

**Comment by jakirkham** [comment]
Wow. Lot's of activity here.

LGTM. @ocefpaf, do you have any more comments?

**Comment by ocefpaf** [comment]
> Wow. Lot's of activity here.

That is why I rebase often, to hide my bad habits of trial-and-error :wink: 

> LGTM. @ocefpaf, do you have any more comments?

Nope. If you are OK merge away!

**Comment by jakirkham** [comment]
Thanks @scopatz.

**Comment by scopatz** [comment]
Thanks a ton for merging this!

**Comment by jakirkham** [comment]
Absolutely, thanks for the addition. If you know anyone else that would be interested in being added as another maintainer here, please feel free to ping them. I'm sure there are some people interested in using `libsigc++`.

**Pull request merged**